db/change_cache.go

```diff
@@ -364,6 +366,16 @@ func (c *changeCache) DocChanged(docID string, docJSON []byte, seq uint64, vbNo
 						Sequence:     seq,
 						TimeReceived: time.Now(),
 					}
+
+					//if the doc was removed from one or more channels at this sequence
+					// Set the removed flag and removed channel set on the LogEntry
+					if channelRemovals, atRevId := doc.Channels.ChannelsRemovedAtSequence(seq); len(channelRemovals) > 0 {
+						change.DocID = docID
+						change.RevID = atRevId
+						change.Flags = channels.Removed
```

Comment:
If change.Flags needed to be set, it would have to be set with |= (or change.setRemoved()), and not =.

However, I don't think this should be set here - it's set for the per-channel Log Entries in _addToCache, based on change.Channels.